File: frontend/src/app/models/post/PostDTO.ts
import { CommunityProfileDTO } from "../communityProfile/CommunityProfileDTO";

export interface PostDTO {
  id: number;
  title: string;
  content: string;
  tags: string[];
  likes: number;
  coments: number;
  communityProfile: CommunityProfileDTO;
  communityId: number;
}

File: backend/src/main/java/com/guildnet/backend/features/post/dto/PostDTO.java
package com.guildnet.backend.features.post.dto;

import com.guildnet.backend.features.communityProfile.CommunityProfile;
import com.guildnet.backend.features.communityProfile.dto.CommunityProfileDTO;
import com.guildnet.backend.features.postComment.dto.PostComentDTO;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PostDTO {
    private Long id;
    private String title;
    private String content;
    private List<String> tags;
    private int likes;
    private int coments;
    private CommunityProfileDTO communityProfile;
    private Long communityId;
}
